gpytorch/utils/function_factory.py

```diff
@@ -26,48 +26,59 @@ def closure(matmul_closure, tr_inv, mat_inv_labels, labels, num_samples):
     return closure
 
 
-def inv_matmul_factory(matmul_closure_factory=_default_matmul_closure_factor, grad_fn=_default_grad_fn):
+def inv_matmul_factory(matmul_closure_factory=_default_matmul_closure_factor,
+                       derivative_quadratic_form_factory=_default_derivative_quadratic_form_factory):
     class InvMatmul(Function):
         def __init__(self, *args):
             self.args = args
 
         def forward(self, *args):
+            if derivative_quadratic_form_factory is None:
```

Comment:
This check should probably be in backward, so that we don't throw an error about missing derivatives unless the user actually calls backward.